src/components/Canvas/Circle.jsx

```diff
@@ -22,8 +26,14 @@ function drawAnimatedCircle (canvas) {
       2 * Math.PI
     )
     ctx.fill()
-    i += 0.05
-    requestId = requestAnimationFrame(draw)
+
+    // if we're NOT paused, update i locally and in parent component
+    if (!isPaused) {
+      i += delta
+      setDrawState({ i })
+    }
+
+    requestId = requestAnimationFrame(() => draw({ isPaused }))
```

Comment:
Because `draw` takes props, we need to update how we recursively call it via `requestAnimationFrame`.

`requestAnimationFrame` calls the provided function with a number (a `DOMHighResTimeStamp`).

https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame

We most likely won't need that, so we wrap the draw call in an anon function